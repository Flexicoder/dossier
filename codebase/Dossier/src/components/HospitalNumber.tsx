import React, { Component, useState } from 'react';
import { Text, TextInput, View } from 'react-native';
import  styles  from '../assets/styles'

interface HospitalNumberProps {
    number: string;
    name: string;
}

export default class HospitalNumber extends Component<HospitalNumberProps> {
    constructor(props: HospitalNumberProps) {
        super(props)
        this.state = { number: "", name: "" }
    }

    render() {
        return (
            <View style={styles.container}>
                <View style={styles.line}>
                    <View style={styles.textBox}>
                    <TextInput
                        placeholder="Hospital Number"
                        onChangeText={text => this.setState({ number: text})}
                        defaultValue={this.props.number} />
                    </View>
                    <View style={styles.textBox}>
                    <TextInput
                        placeholder="Hospital Name"
                        onChangeText={text => this.setState( { name: text })}
                        defaultValue={this.props.name}
                        />
                    </View>
                </View>
            </View>
        );
    }
}