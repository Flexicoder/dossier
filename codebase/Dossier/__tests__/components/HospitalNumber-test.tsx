
import 'react-native';
import React from 'react';
import HospitalNumber from '../../src/components/HospitalNumber';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe("Hospital number", () => {
    it('renders correctly', () => {
        const hospitalNumber = renderer.create(<HospitalNumber name="Brighton" number="123" />);
        expect(hospitalNumber.toJSON()).toMatchSnapshot();
      });
})


describe("Hospital number", () => {
    it('contains supplied data', () => {
        const hospitalNumber = renderer.create(<HospitalNumber name="Brighton" number="123" />);
        const stringVersion = JSON.stringify(hospitalNumber.toJSON())
        expect(stringVersion).toMatch(/Brighton/);
        expect(stringVersion).toMatch(/123/);
      });
})


