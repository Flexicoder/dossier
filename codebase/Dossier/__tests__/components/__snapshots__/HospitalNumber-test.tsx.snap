// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hospital number renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "margin": 5,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue="123"
        onChangeText={[Function]}
        placeholder="Hospital Number"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "borderColor": "lightgrey",
          "borderWidth": 1,
          "color": "grey",
          "flexGrow": 1,
          "fontSize": 18,
          "margin": 5,
          "padding": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue="Brighton"
        onChangeText={[Function]}
        placeholder="Hospital Name"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</View>
`;
